packages/pg-native/index.js

```diff
@@ -1,331 +1,330 @@
-    notice = this.pq.notifies()
-  }
-}
+    // remove instance query error
+    const err = this._queryError
+    this._queryError = undefined
 
-// ensures the client is reading and
-// everything is set up for async io
-Client.prototype._startReading = function () {
-  if (this._reading) return
-  this._reading = true
-  this.pq.on('readable', this._read)
-  this.pq.startReader()
-}
+    // remove instance rows
+    const rows = this._rows
+    this._rows = undefined
+
+    // remove instance results
+    const results = this._results
+    this._results = undefined
 
+    this._resultCount = 0
+
+    if (cb) {
+      cb(err, rows || [], results)
+    }
+  }
+}
 const throwIfError = function (pq) {
   const err = pq.resultErrorMessage() || pq.errorMessage()
   if (err) {
     throw new Error(err)
   }
 }
 
-Client.prototype._awaitResult = function (cb) {
-  this._queryCallback = cb
-  return this._startReading()
-}
-
-// wait for the writable socket to drain
-Client.prototype._waitForDrain = function (pq, cb) {
-  const res = pq.flush()
-  // res of 0 is success
-  if (res === 0) return cb()
-
-  // res of -1 is failure
-  if (res === -1) return cb(pq.errorMessage())
-
-  // otherwise outgoing message didn't flush to socket
-  // wait for it to flush and try again
-  const self = this
-  // you cannot read & write on a socket at the same time
-  return pq.writable(function () {
-    self._waitForDrain(pq, cb)
-  })
-}
-
-// send an async query to libpq and wait for it to
-// finish writing query text to the socket
-Client.prototype._dispatchQuery = function (pq, fn, cb) {
-  this._stopReading()
-  const success = pq.setNonBlocking(true)
-  if (!success) return cb(new Error('Unable to set non-blocking to true'))
-  const sent = fn()
-  if (!sent) return cb(new Error(pq.errorMessage() || 'Something went wrong dispatching the query'))
-  this._waitForDrain(pq, cb)
-}
-
-Client.prototype._onResult = function (result) {
-  if (this._resultCount === 0) {
-    this._results = result
-    this._rows = result.rows
-  } else if (this._resultCount === 1) {
-    this._results = [this._results, result]
-    this._rows = [this._rows, result.rows]
-  } else {
-    this._results.push(result)
-    this._rows.push(result.rows)
-  }
-  this._resultCount++
-}
-
-Client.prototype._onReadyForQuery = function () {
-  // remove instance callback
-  const cb = this._queryCallback
-  this._queryCallback = undefined
-
-  // remove instance query error
-  const err = this._queryError
-  this._queryError = undefined
-
-  // remove instance rows
-  const rows = this._rows
-  this._rows = undefined
-
-  // remove instance results
-  const results = this._results
-  this._results = undefined
-
-  this._resultCount = 0
+// export the version number so we can check it in node-postgres
+module.exports.version = require('./package.json').version
 
-  if (cb) {
-    cb(err, rows || [], results)
-  }
+module.exports = function Client(config) {
```

Comment:
Breaks `instanceof`/inheritance and useful properties like `prototype` and the `version` set on `Client` immediately above as far as existing `const Client = require('pg-native')` users are concerned. I think it’s best to make this a breaking change in a new major version.